extensions/amp-bind/0.1/bind-impl.js

```diff
@@ -587,45 +587,50 @@ export class Bind {
    */
   apply_(results) {
     const applyPromises = this.boundElements_.map(boundElement => {
-      const {element, boundProperties} = boundElement;
 
-      // TODO(choumx): We should avoid triggering a mutation if the expression
-      // results don't affect this element.
-      const applyPromise = this.resources_.mutateElement(element, () => {
+      const boundProperties = boundElement.boundProperties;
+      const propertyUpdates = [];
+      let height, width;
+
+      boundProperties.forEach(boundProperty => {
```

Comment:
Optional: This function is getting pretty long. We can probably split it into two -- calculate updates, then apply them.